vsphere/internal/helper/virtualmachine/virtual_machine_helper.go

```diff
@@ -75,6 +88,64 @@ func FromUUID(client *govmomi.Client, uuid string) (*object.VirtualMachine, erro
 	return vm.(*object.VirtualMachine), nil
 }
 
+// virtualMachineFromSearchIndex gets the virtual machine reference via the
+// SearchIndex MO and is the method used to fetch UUIDs on newer versions of
+// vSphere.
+func virtualMachineFromSearchIndex(client *govmomi.Client, uuid string) (object.Reference, error) {
+	log.Printf("[DEBUG] Using SearchIndex to look up UUID %q", uuid)
+	search := object.NewSearchIndex(client.Client)
+	ctx, cancel := context.WithTimeout(context.Background(), provider.DefaultAPITimeout)
```

Comment:
I think it would be preferable to pass the `ctx` in and declare the `Background` up as high in the callstack as possible, in this case at lease up in `FromUUID`, to make switching to `StopContext` down the road easier.